src/render/src/factory.rs

```diff
@@ -125,51 +125,40 @@ pub trait FactoryExt<R: Resources>: Factory<R> {
         })
     }
 
-    /// Create a constant buffer for `num` identical elements of type `T`.
-    fn create_constant_buffer<T>(&mut self, num: usize) -> handle::Buffer<R, T> {
-        self.create_buffer_dynamic(num, buffer::Role::Constant, Bind::empty())
-            .unwrap()
-    }
-
-    /// Creates and maps a readable buffer.
-    fn create_mapped_buffer_readable<T>(&mut self,
-                                        num: usize,
-                                        role: buffer::Role,
-                                        bind: Bind) -> (handle::Buffer<R, T>,
-                                                        mapping::ReadableOnly<R, T>)
+    /// Creates a constant buffer for `num` identical elements of type `T`.
+    fn create_constant_buffer<T>(&mut self, num: usize)
+                                 -> (handle::Buffer<R, T>,
+                                     mapping::WritableOnly<R, T>)
         where T: Copy
     {
-        let buffer = self.create_buffer_mappable(num, role, bind, memory::READ)
-            .unwrap();
-        let mapping = self.map_buffer_readable(&buffer).unwrap();
-        (buffer, mapping)
+        self.create_upload_buffer(num, buffer::Role::Constant, Bind::empty())
```

Comment:
Yes, I think the constant buffers should be `Data`. Their CPU copies are attached to the command buffers, after all